### PR TITLE
feat(app): devtool(Fallback)ModuleFilenameTemplate can now also be defined as function

### PR DESCRIPTION
**Pull request by thaiat**
Usage example
webpack.config.js
```js
module.exports = {
  entry: 'main.js',
  output: {
    filename : 'bundle.js',
    devtoolModuleFilenameTemplate: function(info) {
      // info is an objet which has the following properties:
      // - identifier
      // - shortIdentifier
      // - resource
      // - resourcePath
      // - absoluteResourcePath
      // - allLoaders
      // - query
      // - moduleId
      // - hash
      return info.resourcePath;
     }
  }
};
```

### CODE REVIEW

**Comment by thaiat** [comment]
Fixes #1222 
Fixes #782

File: lib/ModuleFilenameHelpers.js [hunk 1]
```diff
@@ -66,6 +66,19 @@ ModuleFilenameHelpers.createFilename = function createFilename(module, moduleFil
 	var allLoaders = getBefore(identifier, "!");
 	var query = getAfter(resource, "?");
 	var resourcePath = resource.substr(0, resource.length - query.length);
+	if(typeof moduleFilenameTemplate === "function") {
+		return moduleFilenameTemplate({
+			identifier: identifier,
+            shortIdentifier: shortIdentifier,
```
Comment (sokra) [inline lib/ModuleFilenameHelpers.js]: please use tabs

Comment (thaiat) [inline lib/ModuleFilenameHelpers.js]: done

File: test/configCases/filename-template/module-filename-template/index.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+it("should include test.js in SourceMap", function() {
+    var fs = require("fs");
```
Comment (sokra) [inline test/configCases/filename-template/module-filename-template/index.js]: please use tabs here too

File: test/configCases/filename-template/module-filename-template/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+module.exports = {
+    output: {
+        lineToLine: true,
```
Comment (sokra) [inline test/configCases/filename-template/module-filename-template/webpack.config.js]: and here

Comment (thaiat) [inline test/configCases/filename-template/module-filename-template/webpack.config.js]: done

File: test/configCases/filename-template/module-filename-template/index.js (hunk 2 above)
Comment (thaiat) [inline test/configCases/filename-template/module-filename-template/index.js]: done

**Comment by thaiat** [comment]
any reason you are working with tab instead of spaces ?

**Comment by sokra** [comment]
> any reason you are working with tab instead of spaces ?

yes, but I don't want to start a discussion about tabs vs. spaces here (feel free to google for one). You can see it as personal preference...

**Comment by sokra** [comment]
Thanks

**Comment by urish** [comment]
@thaiat funny, I was missing this feature too and was just about to send a PR, but I see you already did :)

**Comment by thaiat** [comment]
@urish ;)

**Comment by sokra** [comment]
@thaiat Could you add this to the documentation?

**Comment by thaiat** [comment]
@sokra sure, do i modify directly the wiki of the docs repo ?

**Comment by thaiat** [comment]
@sokra i modified the configuration file in the wiki
let me know if this is ok

**Pull request merged**